### ISSUE TITLE
[docs]: NoCloud missing mandatory file information

### ISSUE THREAD

**Issue by stmccartneyncc** [canonical/cloud-init#5858]
# Documentation request
The NoCloud doc page is missing the fact that both user-data and meta-data are required in order for NoCloud to work successfully, and that the other two files are optional.  If you don't have both user-data and meta-data you get the error "device /dev/sd* with label=cidata not a valid seed"

---
*Reported from: https://cloudinit.readthedocs.io/en/latest/reference/datasources/nocloud.html*

**Comment by MarekZydor** [comment]
Just wanted to mention that I've already reported here: [https://github.com/canonical/cloud-init/issues/5843](issue 5843)

**Comment by holmanb** [comment]
To be fair, the [current docs imply](https://cloudinit.readthedocs.io/en/latest/reference/datasources/nocloud.html#source-files) that this is required:

```
For example, if the seedfrom value of seedfrom is https://10.42.42.42/, then the following files will be fetched from the webserver at first boot:

https://10.42.42.42/user-data
https://10.42.42.42/vendor-data
https://10.42.42.42/meta-data
https://10.42.42.42/network-config

If the required files don’t exist, this datasource will be skipped.
```

You're right that the docs don't say _which_ of these are required. I guess we should include that.

**Comment by stmccartneyncc** [comment]
> To be fair, the [current docs imply](https://cloudinit.readthedocs.io/en/latest/reference/datasources/nocloud.html#source-files) that this is required:
> 
> ```
> For example, if the seedfrom value of seedfrom is https://10.42.42.42/, then the following files will be fetched from the webserver at first boot:
> 
> https://10.42.42.42/user-data
> https://10.42.42.42/vendor-data
> https://10.42.42.42/meta-data
> https://10.42.42.42/network-config
> 
> If the required files don’t exist, this datasource will be skipped.
> ```
> 
> You're right that the docs don't say _which_ of these are required. I guess we should include that.

Yes, if that section could be updated to just say that only two of the files are mandatory (and that meta-data can be blank) then I think that would make it much clearer

**Comment by holmanb** [comment]
> and that meta-data can be blank

Could be, perhaps. However this would not be best practice. A unique instance id should be given to each instance.
